ports/rp2/mpnetworkport.c

```diff
@@ -48,6 +48,7 @@ static alarm_id_t lwip_alarm_id = -1;
 #define CYW43_SHARED_IRQ_HANDLER_PRIORITY PICO_SHARED_IRQ_HANDLER_HIGHEST_ORDER_PRIORITY
 
 uint32_t cyw43_country_code = CYW43_COUNTRY_WORLDWIDE;
+int has_pending = 0;
```

Comment:
I think it's better to name this `cyw43_has_pending` and just make it a globally-accessible variable instead of accessing it via a function.  That will be a tiny bit more efficient.

If you want to make a function, I suggest something like this:
```c
extern bool cyw43_has_pending_state;
static inline bool cyw43_has_pending(void) {
    return cyw43_has_pending_state;
}
```

Reply:
And the variable should be `volatile` (thanks @jimmo).